pkg/sr/serde.go

```diff
@@ -172,7 +183,16 @@ func (s *Serde) Register(id int, v any, opts ...SerdeOpt) {
 	// tree to find the end node we are initializing.
 	k := id
 	at := m[k]
-	for _, idx := range t.index {
+	max := func(i, j int) int {
+		if i > j {
+			return i
+		}
+		return j
+	}
+	for i, idx := range t.index {
+		at.subindexDepth = max(at.subindexDepth, len(t.index)-i)
```

Comment:
it'd be clearer to capture `myDepth := len(t.index)` just above the for loop, so this line becomes `at.subindexDepth = max(at.subindexDepth, myDepth)`

Reply:
Can you add a comment above this line:
```
// SAFETY: tserdeMapClone deeply clones all maps through the index,
// so our modified value is being saved to a new map that is not being read.
```